### DISCUSSION TITLE
How to invalidate all generated provider with params

### DISCUSSION THREAD

**Discussion by definev** [rrousselGit/riverpod#2567] [Q&A]
I have a provider to get pagination. When I call `ref.invalidate(getNodesProvider)` the `getNodes disposed` does not print. How to dispose all providers in the `Family` provider?
```dart

@Riverpod(keepAlive: true)
Future<(List<AppNode>, Pagination)> getNodes(
  GetNodesRef ref, {
  required Pagination pagination,
  NodeFilter? filter,
}) async {
  ref.onDispose(() {
    print('getNodes disposed');
  });

  ...
}
```

**Comment by rrousselGit** [comment]
> When I call `ref.invalidate(getNodesProvider)` the `getNodes disposed` does not print.

It should be, assuming the provider was mounted before and not yet disposed.
